databricks_cli/stack/api.py

```diff
@@ -308,6 +321,45 @@ def _deploy_workspace(self, resource_properties, physical_id, overwrite):
 
         return new_physical_id, deploy_output
 
+    def _deploy_dbfs(self, resource_properties, physical_id, overwrite):
+        """
+        Deploy dbfs asset.
+
+        :param resource_properties: dict of properties for the dbfs asset. Must contain the
+        'source_path', 'path' and 'is_dir' fields.
+        :param physical_id: dict containing physical identifier of dbfs asset on Databricks.
+        Should contain the field 'path'.
+        :param overwrite: Whether or not to overwrite the contents of dbfs files.
+        :return: (dict, dict) of (physical_id, deploy_output). physical_id is a dict that
+        contains the dbfs path of the file on Databricks.
+        ex.{"path":"dbfs:/path/in/dbfs"}
+        deploy_output is the initial information about the dbfs asset at deploy time
+        returned by the REST API.
+        """
+        # Required fields. TODO(alinxie) validate fields in _validate_config
+        local_path = resource_properties.get('source_path')
+        dbfs_path = resource_properties.get('path')
+        is_dir = resource_properties.get('is_dir')
+
+        if is_dir != os.path.isdir(local_path):
+            raise StackError("local source_path '{}' is inconsistent"
+                             " with is_dir: {}".format(local_path, is_dir))
+        if is_dir:
+            click.echo('Uploading Directory from {} to Dbfs at {}'.format(local_path, dbfs_path))
+            self.dbfs_client.cp(recursive=True, overwrite=overwrite, src=local_path, dst=dbfs_path)
+        else:
+            click.echo('Uploading File from {} to Dbfs at {}'.format(local_path, dbfs_path))
```

Comment:
micronit: Uploading directory from {} to DBFS at {}.

Reply:
👍